### PR TITLE
Deprecations and warnings cleaned up

### PR DESCRIPTION
**Pull request by alxdarksage**
Updated code to remove deprecations and adjusted the pom.xml file to use the dependencies introduced by Spring Boot when we're not actually overriding them (although we do override a fair number of these dependencies).

I added a comment to the pom.xml file that hides useless pom warnings in Eclipse; I left in lot of @SuppressWarnings annotations that seem to be about suppressing warnings in IntelliJ.

All unit and integration tests pass.

### CODE REVIEW

**Review by jnelsonjava — APPROVED** [review]
Looks good.

File: src/main/java/org/sagebionetworks/bridge/services/SendMailViaAmazonService.java [hunk 1]
```diff
@@ -103,7 +100,6 @@ private void sendEmail(String senderEmail, String recipient, MimeTypeEmail email
         SendRawEmailRequest req = new SendRawEmailRequest(sesRawMessage);
         req.setSource(senderEmail);
         req.setDestinations(Collections.singleton(recipient));
-        emailClient.setRegion(REGION);
         SendRawEmailResult result = emailClient.sendRawEmail(req);
```
Comment (DwayneJengSage) [inline src/main/java/org/sagebionetworks/bridge/services/SendMailViaAmazonService.java:103]: Can we manually verify that Bridge still sends email even if setRegion() is not set?

**Review by DwayneJengSage — APPROVED** [review]
Overall looks good. One comment about setRegion() and SES.

File: src/main/java/org/sagebionetworks/bridge/services/SendMailViaAmazonService.java (hunk 1 above)
Comment (alxdarksage) [reply src/main/java/org/sagebionetworks/bridge/services/SendMailViaAmazonService.java:103]: Yes, it sends email. The deprecation warning on the field says to use the "withRegion()" method of the client builder so I feel pretty confident this is correct.

**Pull request merged**